lib/IDE/TypeReconstruction.cpp

```diff
@@ -2079,49 +2079,6 @@ static void VisitNodeProtocolListWithAnyObject(
   }
 }
 
-static void VisitNodeQualifiedArchetype(
-    ASTContext *ast,
-    Demangle::NodePointer cur_node, VisitNodeResult &result) {
-  if (cur_node->begin() != cur_node->end()) {
-    VisitNodeResult type_result;
-    uint64_t index = 0xFFFFFFFFFFFFFFFF;
-    for (Demangle::NodePointer ChildNd : *cur_node) {
-      switch (ChildNd->getKind()) {
-      case Demangle::Node::Kind::Number:
-        index = ChildNd->getIndex();
-        break;
-      case Demangle::Node::Kind::DeclContext:
-        VisitNode(ast, ChildNd, type_result);
```

Comment:
So I think there are a lot of cases in type reconstruction here that only come up as child nodes of this node that can be removed. But it's going to be hard to figure out what's dead without good test coverage, so let's not attempt it for now.